chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb

```diff
@@ -0,0 +1,67 @@
+#
+# Cookbook Name:: crowbar
+# Recipe:: crowbar-upgrade
+#
+# Copyright 2013-2015, SUSE LINUX Products GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Check for node state: if it is not 'crowbar-upgrading', we're returning
+# from previous upgrade state and need to re-enable chef-client
+# otherwise do the disabling stuff bellow:
+
+if node["state"] == "crowbar-upgrade"
```

Comment:
I'd feel slightly more comfortable using `node["crowbar_wall"]["crowbar-upgrade"]` for the test. I know it doesn't matter, but it's a gut feeling that it's safer in the long term.